charts/nextcloud/templates/nginx-config.yaml

```diff
@@ -9,6 +9,36 @@ metadata:
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 data:
+{{- if .Values.metrics.nginx.enabled }}
+  metrics.conf: |
+    server {
+        listen {{ .Values.metrics.nginx.service.port | default "8081" }};
+
+        # Path to the root of your installation
+        root /var/www/html;
+
+        # Prevent nginx HTTP Server Detection
+        server_tokens off;
+
+        add_header Referrer-Policy                      "no-referrer"   always;
+        add_header X-Content-Type-Options               "nosniff"       always;
+        add_header X-Download-Options                   "noopen"        always;
+        add_header X-Frame-Options                      "SAMEORIGIN"    always;
+        add_header X-Permitted-Cross-Domain-Policies    "none"          always;
+        add_header X-Robots-Tag                         "none"          always;
+        add_header X-XSS-Protection                     "1; mode=block" always;
+
+        # NGINX metrics
+        location /stub_status {
+            stub_status on;
+            allow 127.0.0.1;
+            {{- range .Values.metrics.nginx.allow }}
+                allow {{ . }};
+            {{- end }}
+            deny all;
```

Comment:
it would be nice, that `deny all` could also be disabled (maybe if no allow exists ...)